addons/stock/models/stock_quant.py

```diff
@@ -742,7 +756,10 @@ def _get_available_quantity(self, product_id, location_id, lot_id=None, package_
         :return: available quantity as a float
         """
         self = self.sudo()
-        quants = self._gather(product_id, location_id, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict)
+        if self:
+            quants = self
+        else:
+            quants = self._gather(product_id, location_id, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict)
```

Comment:
```suggestion
        quants = self._gather(product_id, location_id, lot_id=lot_id, package_id=package_id, owner_id=owner_id, strict=strict)
```
Currently it's working but for the future use of the version we could expect people to not filter the record set before calling the funciton with the correct characteristics